airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py

```diff
@@ -93,24 +96,26 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
             if config.start_date and config.end_date < config.start_date:
                 return False, "End date must be equal or after start date."
 
-            api = API(account_id=config.account_id, access_token=config.access_token, page_size=config.page_size)
+            api = API(access_token=config.access_token, page_size=config.page_size)
+
+            for account_id in config.account_ids:
+                # Get Ad Account to check creds
+                ad_account = api.get_account(account_id=account_id)
+                logger.info(f"Select account {ad_account}")
+
+                # make sure that we have valid combination of "action_breakdowns" and "breakdowns" parameters
+                for stream in self.get_custom_insights_streams(api, config):
+                    stream.check_breakdowns(account_id=account_id)
 
-            # Get Ad Account to check creds
-            ad_account = api.account
-            logger.info(f"Select account {ad_account}")
+        except facebook_business.exceptions.FacebookRequestError as e:
```

Comment:
The error should be raised during the check to indicate missing access to one of the accounts. Additionally, during the sync, the error will be raised to avoid silently missing part of the data. Such errors will require the user to fix the connection configuration before proceeding, thus ensuring no data is missed.